File: src/main/java/com/googongill/aditory/service/dto/user/ProfileImageResult.java
package com.googongill.aditory.service.dto.user;

import com.googongill.aditory.domain.User;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class ProfileImageResult {
    private Long userId;
    private String username;
    private String nickname;
    private ProfileImageInfo profileImageInfo;

    public static ProfileImageResult of(User user, ProfileImageInfo profileImageInfo) {
        return ProfileImageResult.builder()
                .userId(user.getId())
                .username(user.getUsername())
                .nickname(user.getNickname())
                .profileImageInfo(profileImageInfo)
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/user/ProfileImageResult.kt
package com.googongill.aditory.service.dto.user

import com.googongill.aditory.domain.User
import com.googongill.aditory.external.s3.dto.S3DownloadResult

data class ProfileImageResult(
    val userId: Long?,
    val username: String,
    val nickname: String?,
    val s3DownloadResult: S3DownloadResult?
) {

    companion object {
        @JvmStatic
        fun of(user: User, s3DownloadResult: S3DownloadResult?): ProfileImageResult {
            return ProfileImageResult(
                userId = user.id,
                username = user.username,
                nickname = user.nickname,
                s3DownloadResult = s3DownloadResult
            )
        }
    }

}
